packages/chart/src/XYAxis.ts

```diff
@@ -389,13 +389,14 @@ export class XYAxis extends SVGWidget {
             .tickLength(this.yAxisGuideLines() ? maxCurrExtent : 0)
             .render()
             ;
-
         this.svgDataClipRect
-            .attr("width", width)
-            .attr("height", height)
+            .attr("width", width - (isHorizontal ? this.xAxis._overlap.left : -this.yAxis._overlap.top))
+            .attr("height", height - (isHorizontal ? this.yAxis._overlap.top : -this.xAxis._overlap.left))
```

Comment:
This code should be moved into "calcMargin" and if possible, should be completely encapsulated inside Axis